src/AbstractWalkers.jl

```diff
@@ -0,0 +1,147 @@
+"""
+Module containing abstract definitions for walkers.
+"""
+module AbstractWalkers
+
+using AtomsBase
+using Unitful
+using ..Potentials
+using ..EnergyEval
+
+export AtomWalker, AtomWalkers, LJAtomWalkers
+export update_walker!
+
+abstract type AtomWalkers end
+
+"""
+    mutable struct AtomWalker
+
+The `AtomWalker` struct represents a walker composed of atoms/molecules.
+
+# Fields
+- `configuration::FastSystem`: The configuration of the walker.
```

Comment:
@yangmr04, if I recall correctly, you looked into two different types of `System`s? Do you think it's worth mentioning that here in some capacity?

Reply:
You mean `FlexibleSystem` vs `FastSystem`? Now in `FreeBird`, only `FlexibleSystem`s are used in actual computations. `FlexibleSystem` is still used but only in I/O related functions so it does not slow the code down. I plan to address this in the documentation (but not necessarily in the function's docstring).